client/jetpack-onboarding/steps/stats.jsx

```diff
@@ -0,0 +1,132 @@
+
+class JetpackOnboardingStatsStep extends React.Component {
+	componentDidUpdate() {
+		const { action, activatedStats, isConnected } = this.props;
+
+		if ( isConnected && activatedStats === false && action === 'activate_stats' ) {
+			this.activateStats();
+		}
+	}
+
+	handleActivateStats = () => {
+		this.props.recordJpoEvent( 'calypso_jpo_stats_clicked' );
+
+		if ( ! this.props.isConnected ) {
+			return;
+		}
+
+		this.activateStats();
+	};
+
+	handleStatsNextButton = () => {
+		this.props.recordJpoEvent( 'calypso_jpo_stats_next_clicked' );
+	};
+
+	activateStats() {
+		this.props.saveJetpackOnboardingSettings( this.props.siteId, {
+			stats: true,
+		} );
+	}
+
+	renderSuccess() {
+		const { getForwardUrl, translate } = this.props;
+		const headerText = translate( 'Success! Jetpack is now collecting valuable stats.' );
+
+		return (
+			<Fragment>
+				<FormattedHeader headerText={ headerText } />
+
+				<TileGrid>
+					<Tile
+						buttonLabel={ translate( 'Continue' ) }
+						image={ '/calypso/images/illustrations/type-business.svg' }
+						onClick={ this.handleStatsNextButton }
+						href={ getForwardUrl() }
+					/>
+				</TileGrid>
+			</Fragment>
+		);
+	}
+
+	renderActionTile() {
+		const { isConnected, siteUrl, translate } = this.props;
+		const headerText = translate( 'Keep track of your visitors with Jetpack.' );
+		const subHeaderText = translate(
+			'Keep an eye on your success with simple, concise, and mobile-friendly stats. ' +
+				'Get updates on site traffic, successful posts, site searches, and comments, all in real time.'
+		);
+		const connectUrl = addQueryArgs(
+			{
+				url: siteUrl,
+				// TODO: add a parameter to the JPC to redirect back to this step after completion
+				// and in the redirect URL include the ?action=activate_stats parameter
+				// to actually trigger the stats activation action after getting back to JPO
+			},
+			'/jetpack/connect'
+		);
+		const href = ! isConnected ? connectUrl : null;
+
+		return (
+			<Fragment>
+				<FormattedHeader headerText={ headerText } subHeaderText={ subHeaderText } />
+
+				<TileGrid>
+					<Tile
+						buttonLabel={ translate( 'Activate stats' ) }
+						image={ '/calypso/images/illustrations/type-business.svg' }
+						onClick={ this.handleActivateStats }
+						href={ href }
+					/>
+				</TileGrid>
+			</Fragment>
+		);
+	}
+
+	render() {
+		const { activatedStats, basePath, translate } = this.props;
+
+		return (
+			<div className="steps__main">
+				<DocumentHead title={ translate( 'Jetpack Stats ‹ Jetpack Start' ) } />
+				<PageViewTracker
+					path={ [ basePath, STEPS.STATS, ':site' ].join( '/' ) }
+					title="Jetpack Stats ‹ Jetpack Start"
+				/>
+
+				{ activatedStats ? this.renderSuccess() : this.renderActionTile() }
+			</div>
+		);
+	}
+}
+
+export default connect(
+	( state, { settings, siteId } ) => ( {
+		activatedStats: get( settings, 'stats' ) === true,
+		isConnected: isJetpackSite( state, siteId ),
```

Comment:
For the `isJetpackSite()` selector to work reliably, you might have to add a `<QuerySites />` component to this step, I think.

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/22621/commits/11d0f20dcc79044c9a32be8681300ce9c12a110a.